src/KeyValueIndex.js

```diff
@@ -10,20 +10,24 @@ class KeyValueIndex {
   }
 
   updateIndex(oplog) {
-    oplog.values
-      .slice()
-      .reverse()
-      .reduce((handled, item) => {
-        if(!handled.includes(item.payload.key)) {
-          handled.push(item.payload.key)
-          if(item.payload.op === 'PUT') {
-            this._index[item.payload.key] = item.payload.value
-          } else if(item.payload.op === 'DEL') {
-            delete this._index[item.payload.key]
-          }
-        }
-        return handled
-      }, [])
+    const values = oplog.values;
+
+    const handled = {};
+    for (let i = values.length - 1; i >= 0; i--) {
+      const item = values[i];
+      if (handled[item.payload.key]) {
```

Comment:
with 1.0 i want to make sure payload is always defined but this is fine for now, all store indexes are are like this atm.